Assets/Scripts/Interaction Scripts/ThirdPersonView.cs

```diff
@@ -0,0 +1,450 @@
+                {
+                    break;
+                }
+
+                case ControllerState.PLACING_WAYPOINT:
+                {
+                    if (controllerInput.GetRightMiddle()) /// Cancel waypoint placement
+                    {
+                        /// TODO: stop showing line
+                        controllerState = ControllerState.IDLE;
+                        RightPointer.enabled = true;
+                        RightPointerRenderer.enabled = true;
+                        break;
+                    }
+
+                    if (!controllerInput.GetRightIndex())
+                    {
+                        controllerState = ControllerState.IDLE;
+                        PlaceWaypoint(RightUI.transform.position);
+                        RightPointer.enabled = true;
+                        RightPointerRenderer.enabled = true;
+                        RightUIPointer.enabled = true;
+                    }
+                    else
+                    {
+                        /// TODO: continue line showing
+                    }
+                    break;
+
+                }
+
+                case ControllerState.MOVING_WAYPOINT:
+                {
+                    break;
+                }
+
+                case ControllerState.UNDOING:
+                {
+                    if (!controllerInput.GetRightA())
+                    {
+                        Undo();
+                        break;
+                    }
+                    if (controllerInput.GetRightMiddle())
+                    {
+                        controllerState = ControllerState.IDLE;
+                        RightPointer.enabled = true;
+                        RightPointerRenderer.enabled = true;
+                        RightUIPointer.enabled = true;
+                        /// TODO: make waypoint disappear
+                        break;
+                    }
+                    controllerState = ControllerState.IDLE;
+                    break;
+                }
+
+                case ControllerState.REDOING:
+                {
+                    if (!controllerInput.GetRightB())
+                    {
+                        Redo();
+                    }
+                    if (controllerInput.GetRightMiddle())
+                    {
+                        controllerState = ControllerState.IDLE;
+                        RightPointer.enabled = true;
+                        RightPointerRenderer.enabled = true;
+                        RightUIPointer.enabled = true;
+                        /// TODO: make waypoint disappear
+                        break;
+                    }
+                    controllerState = ControllerState.IDLE;
+                    break;
+                }
+
+            }
+
+        }
+
+
+        private void ScaleWorld()
+        {
+            // Get the scaling factor, and adjust its size.
+            float ScalingFactor = 1.0f + 0.2f * controllerInput.GetScalingFactor();
```

Comment:
what is the 0.2f?

Reply:
k (constant) to make the scaling factor more natural. arbitrarily chosen